### PR TITLE
Role middleware

### PR DESCRIPTION
**Pull request by bookpanda**
### Type of Change

- [X] New features
- [ ] Bug fixes
- [ ] Breaking changes
- [ ] Refactor
- [ ] Documentation updates
- [ ] Other (please specify):

## Description
- add middleware for superadmin roles
- middleware for admin ALLOWS superadmin but NOT VICE VERSA
- IsLogin = admins + superadmins
- SuperAdmin = superadmins only

### CODE REVIEW

File: internal/interface/handlers/middleware_handler.go [hunk 1]
```diff
@@ -56,3 +58,51 @@ func (h *MiddlewareHandler) IsLogin(c *fiber.Ctx) error {
...
+	h.IsLogin(c)
+
+	userDTO := c.Locals("user").(*dtos.UserDTO)
+	if userDTO.Role != constant.SGCU_ADMIN && userDTO.Role != constant.SGCU_SUPERADMIN {
+		resp := response.NewResponseFactory(response.ERROR, errors.New("Forbidden").Error())
+		return resp.SendResponse(c, fiber.StatusForbidden)
+	}
+
+	return c.Next()
+}
```
Comment (wiraphatys) [inline internal/interface/handlers/middleware_handler.go]: if we use this middleware, no need to put IsLogin() into fiber handler right ?

ex. 
if we want to allow "sgcu_admin" and "sgcu_superadmin" only access route, we just use SGCUAdmin middleware

in the same time if we just want to allow only "sgcu_superadmin" to access just use SGCUSuperAdmin

Comment (bookpanda) [reply internal/interface/handlers/middleware_handler.go]: yes, let me try it first tho555

Comment (macgeargear) [reply internal/interface/handlers/middleware_handler.go]: If so could you fix user router hai noi dai maii 🙇🏻‍♂️ @bookpanda

Comment (wiraphatys) [reply internal/interface/handlers/middleware_handler.go]: or you want the another way kub @bookpanda,
just dont call IsLogin() inside each middleware

Comment (bookpanda) [reply internal/interface/handlers/middleware_handler.go]: ok

**Review by macgeargear — APPROVED** [review]
lgtm

**Comment by bookpanda** [comment]
im not sure whether to make separate middlewares for sccu, sgcu because the routes are the same for both parties right? Maybe we instead check it in usecase

**Review by wiraphatys — APPROVED** [review]
lgtm kub

**Comment by wiraphatys** [comment]
ขอบคุณมากคับสำหรับ makefile ลืมเรื่องนี้ไปเลย

**Pull request merged**